warehouse/query-core/src/main/java/datawave/query/jexl/lookups/ShardIndexQueryTableStaticMethods.java

```diff
@@ -587,44 +599,60 @@ public static RefactoredRangeDescription getRegexRange(Map.Entry<String,String>
                     throws TableNotFoundException, ExecutionException {
         
-        String leadingLiteral = analyzer.getLeadingLiteral(), trailingLiteral = analyzer.getTrailingLiteral();
+        String leadingLiteral = analyzer.getLeadingLiteral();
+        String trailingLiteral = analyzer.getTrailingLiteral();
         
         Set<String> datatypeFilter = config.getDatatypeFilter();
         
-        // Normal case, we have a prefix, use the forward index
+        // TODO Magical handling of a "null" fieldName
+        boolean isForwardIndexed = (null != fieldName) ? indexedInDatatype(fieldName, datatypeFilter, metadataHelper) : true;
+        boolean isReverseIndexed = (null != fieldName) ? reverseIndexedInDatatype(fieldName, datatypeFilter, metadataHelper) : true;
+        
+        // if not indexed at all, then error
+        if (!isForwardIndexed && !isReverseIndexed) {
+            throw new DatawaveFatalQueryException("Cannot lookup a non-indexed term");
+        }
+        
+        // if only indexed one way, then choose that one
+        if (isForwardIndexed != isReverseIndexed) {
+            return isReverseIndexed;
+        }
+        
+        // At this point we know isForwardIndexed == isReverseIndex == true
+        
+        // we only have a prefix, use the forward index
         if (null == trailingLiteral && null != leadingLiteral) {
             return false;
-        } else if (null == leadingLiteral && null != trailingLiteral) {
-            // If we have no prefix, we have no option but to use the reverse index (regardless of indexed or not)
+        }
+        // we only have a suffix, use the reverse index
+        else if (null == leadingLiteral && null != trailingLiteral) {
             return true;
-        } else {
-            // TODO Magical handling of a "null" fieldName
-            boolean isReverseIndexed = (null != fieldName) ? reverseIndexedInDatatype(fieldName, datatypeFilter, metadataHelper) : true;
-            
-            // We have no leading literal, only the trailing, we have to use the reverse index
-            if (null != trailingLiteral && null == leadingLiteral && isReverseIndexed) {
-                return true;
-            }
-            
-            // TODO Magical handling of a "null" fieldName
-            boolean isForwardIndexed = (null != fieldName) ? indexedInDatatype(fieldName, datatypeFilter, metadataHelper) : true;
-            
-            // If we're only reverse indexed, gotta use the reverse
-            if (!isForwardIndexed && isReverseIndexed) {
+        }
+        // we have neither leading or trailing, use foward index
+        if (trailingLiteral == null && leadingLiteral == null) {
+            return false;
+        }
+        // we have both leading and trailing. use the biggest one (sans the realm)
```

Comment:
we might want to default to forward index because the FI is forward indexed only

Reply:
Disregard this, not relevant for global index